### PR TITLE
Make sure to use derived reference in full invariants of abstract values

### PR DESCRIPTION
**Pull request by cblappert**
Release Notes: None

One more invariant fix. 

We used to use `object.property` instead of the `AbstractValue`'s serialized value for invariants. This causes both derived and normal references to be created for the AbstractValue which can cause the AbstractValue to be accessed before it is defined in some cases.

### CODE REVIEW

**Comment by NTillmann** [comment]
The test fails because of an ordering issue. #1449 might fix this.

File: src/utils/generator.js [hunk 1]
```diff
@@ -311,18 +311,19 @@ export class Generator {
...
                 "!==",
-                t.unaryExpression("typeof", accessedPropertyOf(objectNode), true),
+                t.unaryExpression("typeof", valueNode, true),
                 t.stringLiteral(typeString)
               );
             })
           );
           return checks.reduce((expr, newCondition) => t.logicalExpression("&&", expr, newCondition));
         };
+        this.emitInvariant([value, value], condition, valueNode => valueNode);
```
Comment (NTillmann) [inline src/utils/generator.js]: [value]

File: src/utils/generator.js [hunk 2]
```diff
@@ -311,18 +311,19 @@ export class Generator {
...
       }
     } else {
       condition = ([objectNode, valueNode]) => t.binaryExpression("!==", accessedPropertyOf(objectNode), valueNode);
+      this.emitInvariant([object, value, object], condition, objnode => accessedPropertyOf(objnode));
```
Comment (NTillmann) [inline src/utils/generator.js]: Was there a particular reason to list `object` twice, maybe for referencing counting reasons?

**Comment by hermanventer** [comment]
Please fix the failures before requesting a review.

**Pull request closed without merging**